src/Database/Oracle/Simple/FromRow.hs

```diff
@@ -120,3 +120,19 @@ query conn sql = do
         tsVal <- getRow stmt
         found <- fetch stmt
         loop stmt (xs ++ [tsVal]) found
+
+-- | Incrementally process a query
+forEach :: FromRow row => DPIConn -> String -> (row -> IO ()) -> IO ()
```

Comment:
Keeping in line with the new query functions, need to add an underscore since this doesn't accept query parameters and add a `forEach` that does:
```haskell
forEach :: (FromRow a, ToRow b) => Connection -> String -> b -> (a -> IO ()) -> IO ()
forEach conn sql param cont = do
  stmt <- prepareStmt conn sql
  _ <- evalStateT (runRowWriter (toRow param) stmt) (Column 0)
  dpiExecute stmt DPI_MODE_EXEC_DEFAULT
  found <- fetch stmt
  loop stmt found
 where
  loop stmt n | n < 1 = pure ()
  loop stmt _ = do
    tsVal <- getRow stmt
    cont tsVal
    found <- fetch stmt
    loop stmt found
```